stats.go

```diff
@@ -8,18 +8,45 @@ import (
 )
 
 type streamStats struct {
-	messageIn  int
-	messageOut int
-	maxUsers   int
-	start      time.Time
-	mutex      sync.Mutex
-
+	messageIn   int
+	messageOut  int
+	maxUsers    int
+	start       time.Time
+	mutex       sync.Mutex
 	streamStart time.Time
 	streamLive  bool // True if live
+	viewers     map[string]string
+	maxViewers  int
+}
+
+func (s *streamStats) addViewer(ip string) {
+	s.mutex.Lock()
+	s.viewers[ip] = ip
+	s.updateMaxViewers(len(s.viewers))
+	s.mutex.Unlock()
+
+	common.LogDebugf("Viewer connect from: %s\n", ip)
+}
+func (s *streamStats) removeViewer(ip string) {
+	s.mutex.Lock()
+	delete(s.viewers, ip)
+	s.mutex.Unlock()
+
+	common.LogDebugf("Viewer left from: %s\n", ip)
+}
+
+func (s *streamStats) updateMaxViewers(size int) {
+	if s.maxViewers < size {
+		s.maxViewers = size
+	}
+}
+
+func (s *streamStats) resetViewers() {
+	s.viewers = make(map[string]string)
 }
 
 func newStreamStats() streamStats {
-	return streamStats{start: time.Now(), streamLive: false}
+	return streamStats{start: time.Now(), streamLive: false, viewers: make(map[string]string)}
```

Comment:
I know its kinda "late" for this question since the implementation is already mostly done but why do you use a map here instead of a simple slice/array?